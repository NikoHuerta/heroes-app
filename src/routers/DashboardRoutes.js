import React from 'react'
import {
    Routes,
    Route
  } from 'react-router-dom';
import { DcScreen } from '../components/dc/DcScreen';
import { HeroScreen } from '../components/heroes/HeroScreen';
import { MarvelScreen } from '../components/marvel/MarvelScreen';
import { Navbar } from '../components/ui/Navbar';

export const DashboardRoutes = () => {
    return (
        <>
            <Navbar />   
            <div className="container mt-2">
                <Routes>
                    <Route exact path="/marvel" element={ <MarvelScreen /> }/>
                    <Route exact path="/hero/:heroeId/*" element={ <HeroScreen /> }/>
                    <Route exact path="/dc" element={ <DcScreen /> }/>

                    <Route path="*" element={ <MarvelScreen /> } />
                </Routes>
            </div>
        </>
    )
}
