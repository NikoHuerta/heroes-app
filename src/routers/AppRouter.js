import React from 'react'
import {
    BrowserRouter as Router,
    Routes,
    Route
  } from 'react-router-dom';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';

// import { Navbar } from '../components/ui/Navbar';



export const AppRouter = () => {
    return (
    <Router>
      <div>
        {/* <Navbar /> */}

        <Routes>
          <Route exact path="/login" element={ <LoginScreen /> }/>
          <Route exact path="/*" element={ <DashboardRoutes /> }/>
        </Routes>
        
      </div>
    </Router>
    );
}
