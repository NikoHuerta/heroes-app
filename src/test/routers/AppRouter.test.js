import { mount } from "enzyme";
import { AuthContext } from "../../auth/authContext";
import { AppRouter } from "../../routers/AppRouter";

describe('Pruebas en <AppRouter />', () => {
   
    const contextValue = {
        user:{
            logged: false
        }
    };
    
    
    test('Debe de mostrar el Login si no esta autenticado', () => {
        
        const contextValue = {
            user:{
                logged: false
            }
        };
       
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('h1').text().trim()).toBe('Login Screen');
    });
    
    test('Debe de mostrar el componente SearchScreen si esta autenticado', () => {
        
        const contextValue = {
            user:{
                logged: true,
                name: 'Nicolas',
                email: 'nico@las.h'
            }
        };
       
        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <AppRouter />
            </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.navbar').exists()).toBe(true);
    });


});
