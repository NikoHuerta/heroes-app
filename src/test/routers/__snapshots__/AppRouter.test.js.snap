// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppRouter /> Debe de mostrar el Login si no esta autenticado 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
       Login Screen 
    </h1>
    <hr />
    <form
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        className="form-control"
        name="nameLogin"
        onChange={[Function]}
        placeholder="Enter your name"
        type="text"
        value=""
      />
      <br />
      <button
        className="btn btn-primary"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;

exports[`Pruebas en <AppRouter /> Debe de mostrar el componente SearchScreen si esta autenticado 1`] = `
<div>
  Array [
    <nav
      className="navbar navbar-expand-sm navbar-dark bg-dark"
    >
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Asociaciones
      </a>
      <div
        className="navbar-collapse"
      >
        <div
          className="navbar-nav"
        >
          <a
            className="nav-item nav-link"
            href="/marvel"
            onClick={[Function]}
          >
            Marvel
          </a>
          <a
            className="nav-item nav-link"
            href="/dc"
            onClick={[Function]}
          >
            DC
          </a>
          <a
            aria-current="page"
            className="nav-item nav-link active"
            href="/search"
            onClick={[Function]}
          >
            Search
          </a>
          <a
            className="nav-item nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </div>
      </div>
      <div
        className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
      >
        <ul
          className="navbar-nav ms-auto"
        >
          <span
            className="nav-item nav-link text-info"
          >
            Nicolas
          </span>
          <button
            className="nav-item nav-link btn"
            onClick={[Function]}
          >
            Logout
          </button>
        </ul>
      </div>
    </nav>,
    <div
      className="container mt-2"
    >
      <div>
        <h1>
           Search Screen
        </h1>
        <hr />
        <div
          className="row"
        >
          <h4>
            Search Form
          </h4>
          <hr />
          <form
            onSubmit={[Function]}
          >
            <input
              autoComplete="off"
              className="form-control"
              name="searchString"
              onChange={[Function]}
              placeholder="Find Your Hero"
              type="text"
              value=""
            />
            <button
              className="btn m-1 btn-block btn-outline-primary"
              type="submit"
            >
              Search
            </button>
          </form>
        </div>
        <div>
          <br />
          <h4>
            Results
          </h4>
          <hr />
          <br />
          <div
            className="alert alert-info"
          >
            Search a Hero
          </div>
          <div
            className="row row-cols-1 row-cols-md-3 g-4"
          />
        </div>
      </div>
    </div>,
  ]
</div>
`;
